TUTORIAL.md

```diff
@@ -0,0 +1,245 @@
+#Guide to Solving and Reviewing Rails Blogs Nested Forms
+##Overview
+This lab will teach you how to implement nested forms. In this example, we will nest tags under posts, so a user can submit a post that has many tags.
+
+###Step 1: Set up your app
+- `bin/rake db:migrate RAILS_ENV=test`
+- `bundle install`
+- `rails s` then visit `localhost:3000`
+- `rake db:seed`
+
+At this point we should have a form that will let us create a post, along with tags that we can select. Now we need a way to add a new tag, so let's add that to our form.
+
+###Step 2: Use fields_for to accept tags
+
+```ruby
+<div class="field">
+  <%= f.fields_for(:tags) do |tag_form| %>
+    <%= tag_form.label :name %>
+    <%= tag_form.text_field :name %>
+  <% end %>
+</div>
+
+```
+####Completed Form
+
+```ruby
+<%= form_for(@post) do |f| %>
+  <% if @post.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(@post.errors.count, "error") %> prohibited this post from being saved:</h2>
+      <ul>
+      <% @post.errors.full_messages.each do |msg| %>
+        <li><%= msg %></li>
+      <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field">
+    <%= f.label :name %><br>
+    <%= f.text_field :name %><br>
+    <%= f.label :content %><br>
+    <%= f.text_area :content %><br>
+    Tags:
+  </div>
+
+  <div class="field">
+    <%= f.fields_for(:tags) do |tag_form| %>
+      <%= tag_form.label :name %>
+      <%= tag_form.text_field :name %>
+    <% end %>
+  </div>
+
+  <div class="field">
+    <%= f.collection_check_boxes :tag_ids, Tag.all, :id, :name %><br>
+  </div>
+
+  <div class="actions">
+    <%= f.submit %>
+  </div>
+
+<% end %>
+```
+
+####Before we go any further, let's break down what is happening above. 
+
+Our `form_for` accepts `@post` as an argument, which is an instance of our `Post` class. We then pass it to the variable `f` to build our form.
+
+Let's take a look at `@post` params in our console and see what is being returned. 
+
+
+```ruby
+#<Post:0x007fb3ad9e4e50> {
+            :id => 10,
+          :name => "Hello World!",
+       :content => "We are learning about fields_for!",
+       :user_id => nil,
+    :created_at => Thu, 09 Jul 2015 16:59:41 UTC +00:00,
+    :updated_at => Thu, 09 Jul 2015 16:59:41 UTC +00:00
+}
+```
+When we call `<%= f.label :name %>` in our form above, we are accessing the `:name` key in our hash, which is going to return the name of that post, same thing for `:content`, or any other key that is available in our params.
```

Comment:
If you take a look, that isn't a hash. It's a Post object. (you can tell because of the `#<Post:0x007fb3ad9e4e50>` stuff) So you aren't accessing the `:name` key, you are calling the `name` method on the instance of the Post object.